Assignment1/renderer.cpp

```diff
@@ -139,4 +121,66 @@ void Tmpl8::Renderer::KeyDown(int key)
 	if (key == 0x44) mov += float3(1, 0, 0);; //D
 	if (key == VK_SPACE) mov += float3(0, 1, 0); // Space bar
 	if (key == 0x43) mov += float3(0, -1, 0); // C
+	if (key == 0x45) fovc += float3(-1, 0, 0);// E
+	if (key == 0x51) fovc += float3(1, 0, 0);// Q
+}
+
+float3 Tmpl8::Renderer::Whitted(float3 I,float3 N,  Ray & ray, Material& m)
+{
+	float d = m.diffuse; 
+	float s = m.specular;
+
+	float3 distance_to_light = scene.lights[0].position - I;
+	float3 dirtolight = normalize(distance_to_light);
+	Ray occlusion_ray = Ray(I + 0.0002f * dirtolight, dirtolight, length(distance_to_light), ray.objIdx+1);
+
+	// Whitted Tracing --> uncomment 
+if (ray.depthidx > max_depth) {
+	// at maximum depth we try to return the last object hit's color, or just darkness for "eternal reflection"
+
+if (scene.IsOccluded(occlusion_ray)) return float3(0.0f, 0.0f, 0.0f); }						// occlusion
+if (!scene.IsOccluded(occlusion_ray)) {
+	if (d > 0.0) return  d * scene.directIllumination(ray.objIdx, I, N, m.albedo);			// If diffuse
+	if (s > 0.0)  return  s * Trace(ray.Reflect(I, N));										// If specular -> // Only if we hit front of the material
```

Comment:
This should not be in the if statement. Reflections work even if the mirror is occluded.